src/blockdata/script.rs

```diff
@@ -395,6 +401,14 @@ impl Script {
             self.0[1] == opcodes::all::OP_PUSHBYTES_20.into_u8()
     }
 
+    /// Checks whether a script pubkey is a p2tr output
+    #[inline]
+    pub fn is_v1_p2tr(&self) -> bool {
+        self.0.len() == 34 &&
+        self.0[0] == opcodes::all::OP_PUSHNUM_1.into_u8() &&
+        self.0[1] == opcodes::all::OP_PUSHBYTES_32.into_u8()
```

Comment:
Not related to the PR, but just a thought that we probably need to refactor all these opcode-based version guessing via WitnessVersion API